### PR TITLE
 kubevirt: add Clone Dialog to actions

### PR DESCRIPTION
**Pull request by suomiy**
depends on:
- [x] #1819

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx [hunk 1]
```diff
@@ -136,12 +148,58 @@ const menuActionCancelMigration = (
...
+        getResource(PersistentVolumeClaimModel, { namespace, prop: 'persistentVolumeClaims' }),
+        getResource(DataVolumeModel, { namespace, prop: 'dataVolumes' }),
+      ];
+
+      const resourcesToProps = (props) => ({
+        vm,
+        ...resources
+          .map((r) => r.prop)
+          .reduce((mappedResources, propName) => {
+            mappedResources[propName] = getLoadedData(props[propName]);
```
Comment (mareklibra) [inline frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx]: I think we should handle the case when the user has not enough privileges to list these resources.

File: frontend/packages/kubevirt-plugin/src/utils/index.ts [hunk 2]
```diff
@@ -30,3 +30,8 @@ export const createLookup = <A extends K8sResourceKind>(
 
 export const prefixedId = (idPrefix: string, id: string): string =>
   idPrefix && id ? `${idPrefix}-${id}` : null;
+
+export const getLoadedData = (
+  result: FirehoseResult<K8sResourceKind | K8sResourceKind[]>,
+  defaultValue = null,
+) => (result && result.loaded ? result.data : defaultValue);
```
Comment (mareklibra) [inline frontend/packages/kubevirt-plugin/src/utils/index.ts]: the `loaded` is `true` even if the load failed (like HTTP 403 or timeout)

**Comment by mareklibra** [comment]
Needs rebase.
We should handle error-cases during data loading but this can be done in a follow-up.
Otherwise lgtm.

**Comment by mareklibra** [comment]
@rawagner, thanks for the rebase.

@suomiy , I do not want block merge of this but the error handling needs to be resolved before development freeze, can you please have a look?

/lgtm

**Comment by mareklibra** [comment]
/lgtm cancel

Recent build fails on:
```

ERROR in /go/src/github.com/openshift/console/frontend/packages/kubevirt-plugin/src/components/vms/menu-actions.tsx
(11,10): Module '"../../../../../node_modules/@console/shared/src"' has no exported member '``'.
```

@rawagner, can you please fix? It was moved to `frontend/packages/kubevirt-plugin/src/components/modals/modal-resource-launcher.tsx`.

**Comment by suomiy** [comment]
> @suomiy , I do not want to block merge of this but the error handling needs to be resolved before development freeze anyway, can you please have a look?

I believe this has to be implemented in the dialog itself.

@jelkosz priority?

**Comment by jelkosz** [comment]
> @jelkosz priority?

before freeze yes but do not block this PR on it.

**Comment by suomiy** [comment]
Added default value so it will not break when missing the data. Error handling will be added in a followup

**Comment by jelkosz** [comment]
restoring
/lgtm

**Pull request merged**